app/routes.py

```diff
@@ -1,2 +1,234 @@
-from flask import Blueprint
+from app import db
+import requests
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import request, Blueprint, make_response, jsonify
+from sqlalchemy import asc, desc
+from datetime import datetime
+import os
+from dotenv import load_dotenv
+from dateutil.parser import parse
 
+load_dotenv()
+
+#################### Routes for Tasks ####################
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def get_tasks():
+    order_query = request.args.get("sort")
+    order_by_id = request.args.get("sort_by_id")
+    filter_by_title = request.args.get("filter_by_title")
+    if order_query:
+        if order_query == "asc":
+            tasks = Task.query.order_by(asc(Task.title))
+        elif order_query == "desc":
+            tasks = Task.query.order_by(desc(Task.title))
+        else: 
+            raise TypeError("Only asc or desc is accepted here")
+    elif order_by_id:
+        if order_by_id == "asc":
+            tasks = Task.query.order_by(asc(Task.task_id))
+        elif order_by_id == "desc":
+            tasks = Task.query.order_by(desc(Task.task_id))
+        else: 
+            raise TypeError("Only asc or desc is accepted here")
+    elif filter_by_title:
+        tasks = Task.query.filter_by(title=filter_by_title).all()
+    else:
+        tasks = Task.query.all()
+    tasks_response = [task.to_json() for task in tasks]
+    return jsonify(tasks_response), 200
+  
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def get_single_task(task_id):
+    task = Task.query.get(task_id)
+    if task:
+        return {"task": task.to_json()}, 200
+    else:
+        return jsonify(None), 404
+      
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_task():
+    request_body = request.get_json()
+    if all(key in request_body for key in ("title", "description", "completed_at")):
+        if validate_data(request_body): 
```

Comment:
Just a note that your `validate_data` function raises an exception if there's a problem, so it wouldn't reach the `else` block.